examples/starter-policy/fastapi/sample_application.py

```diff
@@ -0,0 +1,216 @@
+import oso_sdk
+import os
+from fastapi import Depends, FastAPI, Request
+from fastapi.exceptions import HTTPException
+from oso_sdk.integrations.fastapi import FastApiIntegration
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ApiKeyNotFound(Exception):
+    """
+    Unable to start application, could not retrieve Oso Cloud API key.
+    """
+
+    pass
+
+
+def api_key() -> str:
+    """
+    Retrieve API key, checking `OSO_AUTH` environment variable.
+
+    Raise an exception if no value is set.
+    """
+    api_key = os.environ.get("OSO_AUTH")
+    if api_key is None:
+        raise ApiKeyNotFound
+    else:
+        return api_key
+
+
+oso = oso_sdk.init(
+    api_key(),
+    FastApiIntegration(),
+    # create a local instance of the Oso SDK
+    shared=False,
+    # only enforce routes with the `@oso.enforce` decorator
+    optin=True,
+)
+
+app = FastAPI(dependencies=[Depends(oso)])
+
+
+@oso.identify_user_from_request
+async def user(request: Request) -> str:
+    """
+    Oso Cloud handles authorization - you'll need to handle your own
+    authentication. That means it's up to you to figure out *who* a user is
+    (for example, with a username and password).
+
+    For this sample application, we identify the user by checking
+    the request's Authorization header.
+
+    For pedagogical purposes, we assume there are effectively two users:
+    - "admin"
+    - "anonymous"
+
+    If the value of the Authorization header is "secret_password", we assume the
```

Comment:
This is difficult to achieve outside of `cURL` and `postman` environments, so we should consider something else. Either use `Basic` auth (see oso-by-example) or skip this altogether.

If we tie this back to the demo environment, we could potentially have this return a static value. This the reason I moved away from having the default be a JWT in the SDK.

Reply:
this sample application now only has one user (`User{"anonymous"}`), and does not use any `Authorization` headers